### PR TITLE
Fixes in iAbstractCloud interface

### PR DESCRIPTION
**Pull request by ancord-m**

### CODE REVIEW

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+     * An uploadResource method implementation takes as a parameter a path to a resource
+     * which is stored at user's local HDD/SSD and sends it to an external cloud storage.
+     * Notice, that resource may be a file or directory with files.
+     *
+     * @param path A path to a resource at user's local storage.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+
+    String uploadResource(String path);
```
Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: If you have resource type is file or directory, may be use parameter as Java type File?

What is return vales of String type? What contain this String?

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    /**
+     * Provides the ability to upload a resource from one cloud disk storage to another.
+     * A method should somehow request a URL for downloading and pass it as parameter.
+     * Also a URL may point to any file that is available at the Internet.
+     * @param path A path where the uploaded resource should be stored.
+     * @param url A URL to an external resource.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String uploadExternalResource(String path, String url);
```
Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: I think that we should be change types of parameters (path to File or Path, url as URL)

Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: And I think that we should use some object as return value. We can create abstract class JsonObject (if  we have dependencies from JSON view) and child classes ExternalResource.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    /**
+     * A downloadBinFile method implementation is responsible for downloading a file-chunk from
+     * a cloud storage to user's local storage.
+     * @param path
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String downloadResource(String path);
```
Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: If you use abstraction Resource for parent entity for File and Directory, you should be create CRUD methods and managment metods(copy, move, get/set metadata)  for Resource entity.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+     * Depending on a cloud storage provider the response may vary.
+     * At least the returned value should contain:
+     * - total space available
+     * - used space
+     * = free space
+     *
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String getCloudDiskMetadata();
```
Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: And in this interface we should be have methods for connect\disconnect, authenticate on cloud.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java (hunk 1 above)
Comment (ancord-m) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: 1. Do you mean uploadResource(File path) ?
2. I thought that the String as the return type may contain external cloud server responce. I've looked through Dropbox / Yandex.Disk documentation and saw that responce includes may useful information (size, date, metadata, download link and so on..). All this information may be stored in String object and then parsed by means of an external helper class which will give us a JSON object. But, as you have told me, we can return a JSON object and thus omit external String -> JSON object conversion.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java (hunk 3 above)
Comment (ancord-m) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: Firstly, there were "...File..." and "...Directory...", but it seems to me, that almost in all cases there is no difference between operation with a file or a directory. That's why I've unified them and named Resource. On the other hand we are implementing REST and CRUD are four main methods.. So do you want me to rename methods?

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java (hunk 4 above)
Comment (ancord-m) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: Give me a piece of advice, please: how can I do this?

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java (hunk 3 above)
Comment (slonikmak) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: I also think that the best way is to use the same method for files and directories in Java nio. It is easier to use and there is no need for the users of the class that implements the interface, do the validation and to choose the right methods. For identification of file I propose to use the Path class everywhere

**Comment by kosmoflyko** [comment]
fix #41

**Pull request merged**